effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -778,7 +778,7 @@ object Typer extends Phase[NameResolved, Typechecked] {
 
   def synthDef(d: Def)(using Context, Captures): Result[Unit] = Context.at(d) {
     d match {
-      case d @ source.FunDef(id, tps, vps, bps, ret, body, doc, span) =>
+      case d @ source.FunDef(id, tps, vps, bps, cpt, ret, body, doc, span) =>
         val sym = d.symbol
         // was assigned by precheck
         val functionCapture = Context.lookupCapture(sym)
```

Comment:
Since you already allow annotating captures on non-extern functions, here we might want to implement the check. That is, conceptually `functionCapture <: cpt`.

Reply:
On the other side, in Namer `functionCapture` could be (and maybe is already?) be instantiated to be `cpt` instead of a capture variable.